src/scheduling/schedule_config.py

```diff
@@ -32,6 +31,37 @@ def _read_repeat_period(incoming_schedule_config):
     return period
 
 
+def _read_executions_count(incoming_schedule_config):
+    executions_count = model_helper.read_int_from_config('executions_count', incoming_schedule_config)
+    if executions_count is None:
+        executions_count = 0
+    elif executions_count < 0:
+        raise InvalidScheduleException('executions_count should be > 0')
+    return executions_count
+
+def _read_end_arg_int(incoming_schedule_config):
+    end_arg = model_helper.read_int_from_config('end_arg', incoming_schedule_config)
+    if end_arg is None:
+        raise InvalidScheduleException('end_arg is required for repeatable schedule')
+    elif end_arg <= 0:
+        raise InvalidScheduleException('end_arg should be > 0')
+    return end_arg
+
+
+def _read_end_args(incoming_schedule_config):
+    end_option = incoming_schedule_config.get('end_option')
+    if end_option is None:
+        raise InvalidScheduleException('end_option is required for repeatable schedule')
```

Comment:
I think None `end_option` should be allowed:
- existing configurations don't have it
- we should make manual configuration (via json) more user-friendly, i.e. don't require unnecessary configurations